packages/react-dom/src/client/DOMPropertyOperations.js

```diff
@@ -141,8 +161,13 @@ export function setValueForProperty(
       const attributeName = name;
       if (value === null) {
         node.removeAttribute(attributeName);
+      } else if (__DEV__) {
+        node.setAttribute(
+          attributeName,
+          stringifyWithPerformanceWarning(value),
```

Comment:
we could skip doing this with the warning if value.toString belongs to one of the primitives (Number, Boolean, String etc)

Reply:
do you mean, directly doing `warning()` rather that `stringifyWithPerformanceWarning()`